### PR TITLE
Client + docs: rename `Program`

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Client + docs: rename `Program`

### CODE REVIEW

File: client/quantum_serverless/core/program.py [hunk 1]
```diff
@@ -72,21 +74,58 @@ class Program:  # pylint: disable=too-many-instance-attributes
...
+            version,
+            tags,
+            raw_data,
+        )
+
+    def __repr__(self):
+        return f"Program({self.title})"
+
+
 class ProgramStorage(ABC):
```
Comment (psschwei) [inline client/quantum_serverless/core/program.py:125]: Do we need to `s/Program/Pattern/` here (and elsewhere in this class, i.e. `save_program` and the like)?
Or :shudder: change the filename itself from program to pattern?

Comment (IceKhan13) [reply client/quantum_serverless/core/program.py:87]: good point!

**Review by psschwei — APPROVED** [review]
Should this also get tagged as a feature branch? there's a new class being added / an old one being deprecated, so _technically_ ... (?)

File: docs/getting_started/basic/02_arguments_and_results.ipynb [hunk 2]
```diff
@@ -186,20 +186,9 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": null,
```
Comment (psschwei) [inline docs/getting_started/basic/02_arguments_and_results.ipynb]: do you want to execute this cell? (looks like the rest of them were execute...)

File: docs/getting_started/experimental/running_programs_using_decorators.ipynb [hunk 3]
```diff
@@ -216,65 +216,43 @@
...
     "\n",
-    "job = my_program_with_modules()\n",
+    "job = my_pattern_with_modules()\n",
     "job"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 7,
+   "execution_count": null,
```
Comment (psschwei) [inline docs/getting_started/experimental/running_programs_using_decorators.ipynb:252]: do you want to execute this cell? (think there might also one other in this notebook that wasn't executed...)

File: client/quantum_serverless/core/job.py [hunk 4]
```diff
@@ -166,11 +168,13 @@ def run(self, program: Program, arguments: Optional[Dict[str, Any]] = None):
         )
         return Job(job_id=job_id, job_client=self)
 
-    def upload(self, program: Program):
+    def upload(self, program: QiskitPattern):
```
Comment (Tansito) [inline client/quantum_serverless/core/job.py:171]: This can a breaking change so it's a bit tricky but should we call it `pattern`?

**Review by Tansito — APPROVED** [review]
I left a couple of comments and I found a couple of references in the documentation that maybe we can review but I would merge it and do next week another pass, what do you think, @IceKhan13 ?

**Comment by IceKhan13** [comment]
> Should this also get tagged as a feature branch? there's a new class being added / an old one being deprecated, so technically ... (?)

This is more of a docs change + name alias. It does not change any functionality

File: client/quantum_serverless/core/job.py (hunk 4 above)
Comment (psschwei) [reply client/quantum_serverless/core/job.py:171]: can you overload functions in python? (i.e. have one that takes a program and mark it deprecated and another that takes a pattern)

Comment (Tansito) [reply client/quantum_serverless/core/job.py:171]: good question!

**Comment by IceKhan13** [comment]
We will follow with updates in next PR :)

File: client/quantum_serverless/core/job.py (hunk 4 above)
Comment (psschwei) [reply client/quantum_serverless/core/job.py:171]: seems like answer is no (though there are a few workarounds if you search around on the topic...)

**Pull request merged**